subsys/bluetooth/services/ias.c

```diff
@@ -0,0 +1,125 @@
+/** @file
+ *  @brief Immediate Alert Service implementation
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <stdint.h>
+#include <zephyr.h>
+#include <net/buf.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/ias.h>
+
+#define LOG_LEVEL CONFIG_BT_IAS_LOG_LEVEL
+LOG_MODULE_REGISTER(ias);
+
+#define BT_IAS_ALERT_MAX_LEN 1
+
+#if defined(CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN)
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE_AUTHEN
+#elif defined(CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT)
+#define IAS_GATT_PERM_DEFAULT CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT 
+#else
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE
+#endif
+
+typedef enum __packed {
+	BT_IAS_ALERT_LVL_NO_ALERT,
+	BT_IAS_ALERT_LVL_MILD_ALERT,
+	BT_IAS_ALERT_LVL_HIGH_ALERT,
+} bt_ias_alert_lvl_t;
+
+bt_ias_alert_lvl_t alert_level;
+
+static void set_alert_level(bt_ias_alert_lvl_t alert_lvl)
+{
+	if (alert_lvl == alert_level) {
+		return;
+	}
+
+	if (alert_lvl == BT_IAS_ALERT_LVL_NO_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb) {
+			if (cb->stop_alert) {
+				cb->stop_alert();
+				alert_level = BT_IAS_ALERT_LVL_NO_ALERT;
+				LOG_DBG("Alert stop");
+			}
+		}
+	} else if (alert_lvl == BT_IAS_ALERT_LVL_MILD_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb) {
+			if (cb->start_alert) {
+				cb->start_alert();
+				alert_level = BT_IAS_ALERT_LVL_MILD_ALERT;
+				LOG_DBG("Mild alert start");
+			}
+		}
+	} else if (alert_lvl == BT_IAS_ALERT_LVL_HIGH_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb) {
+			if (cb->start_alert_high) {
+				cb->start_alert_high();
+				alert_level = BT_IAS_ALERT_LVL_HIGH_ALERT;
+				LOG_DBG("High alert start");
+			}
+		}
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (alert_level != BT_IAS_ALERT_LVL_NO_ALERT) {
+		set_alert_level(BT_IAS_ALERT_LVL_NO_ALERT);
+	}
+}
+
+void local_alert_stop()
+{
+	STRUCT_SECTION_FOREACH(bt_ias_cb, cb) {
+		if (cb->stop_alert) {
+			cb->stop_alert();
+			alert_level = BT_IAS_ALERT_LVL_NO_ALERT;
+			LOG_DBG("Alert stop");
+		}
+	}
```

Comment:
```suggestion
	if (alert_level != BT_IAS_ALERT_LVL_NO_ALERT) {
		set_alert_level(BT_IAS_ALERT_LVL_NO_ALERT);
	}
```

Basically the `local_alert_stop` and `disconnected` should work the same.